sim/pokemon.ts

```diff
@@ -1126,7 +1126,9 @@ export class Pokemon {
 		const species = pokemon.species;
 		if (pokemon.fainted || pokemon.illusion || (pokemon.volatiles['substitute'] && this.battle.gen >= 5) ||
 			(pokemon.transformed && this.battle.gen >= 2) || (this.transformed && this.battle.gen >= 5) ||
-			species.name === 'Eternatus-Eternamax') {
+			species.name === 'Eternatus-Eternamax' || (this.battle.dex.currentMod === 'gen1stadium' && species.name === 'Ditto') ||
+      (this.battle.dex.currentMod === 'gen1stadium' && this.species.name === 'Ditto' &&
+      pokemon.moves.includes('transform'))) {
```

Comment:
This is a bit of an unnecessarily huge if statement. Let's just put this in another if statement?

Reply:
Yeah it is, I don't mind splitting it off from the other stuff